packages/liquidator/src/proxyTransactionWrapper.js

```diff
@@ -0,0 +1,288 @@
+const assert = require("assert");
+const truffleContract = require("@truffle/contract");
+
+const ynatm = require("@umaprotocol/ynatm");
+
+const { createObjectFromDefaultProps, MAX_UINT_VAL } = require("@uma/common");
+const { getAbi, getTruffleContract } = require("@uma/core");
+
+const UniswapV2Factory = require("@uniswap/v2-core/build/UniswapV2Factory.json");
+const IUniswapV2Pair = require("@uniswap/v2-core/build/IUniswapV2Pair.json");
+
+class ProxyTransactionWrapper {
+  /**
+   * @notice Constructs new ProxyTransactionWrapper. This adds support DSProxy atomic liquidation support to the bots.
+   * @param {Object} web3 Provider from Truffle instance to connect to Ethereum network.
+   * @param {Object} financialContract instance of a financial contract. Either a EMP or a perp. Used to send liquidations.
+   * @param {Object} gasEstimator Module used to estimate optimal gas price with which to send txns.
+   * @param {Object} syntheticToken Synthetic token issued by the financial contract(tokenCurrency).
+   * @param {Object} collateralToken Collateral token backing the financial contract.
+   * @param {String} account Ethereum account from which to send txns.
+   * @param {Object} dsProxyManager Module to send transactions via DSProxy. If null will use the unlocked account EOA.
+   * @param {Boolean} isUsingDsProxyToLiquidate Toggles the mode liquidations will be sent with. If true then then liquidations.
+   * are sent from the DSProxy. Else, Transactions are sent from the EOA. If true dsProxyManager must not be null.
+   * @param {Object} proxyTransactionWrapperConfig configuration object used to paramaterize how the DSProxy is used. Expected:
+   *      { uniswapRouterAddress: 0x123..., // uniswap router address. Defaults to mainnet router
+            uniswapFactoryAddress: 0x123..., // uniswap factory address. Defaults to mainnet factory
+            liquidatorReserveCurrencyAddress: 0x123... // address of the reserve currency for the bot to trade against
+            maxReserverTokenSpent: "10000" // define the maximum amount of reserve currency the bot should use in 1tx. }
+   * */
+  constructor({
+    web3,
+    financialContract,
+    gasEstimator,
+    syntheticToken,
+    collateralToken,
+    account,
+    dsProxyManager = undefined,
+    isUsingDsProxyToLiquidate = false,
```

Comment:
Maybe i'm trying to over-refactor, but why can't we remove the `isUsingDsProxyToLiquidate` config boolean, and just check if `dsProxyManager == undefined`? It seems below on lines 94, 122,  and 182 that you are checking the former but I feel like you can also just check for `!dsProxymanager`

Reply:
you could do that. the reason I did it like this is that I want the user to explicitly define a `useDsProxyToLiquidate=true` within the `DSPROXY_CONFIG` and by doing it like this I can simply pass down the `proxyTransactionWrapperConfig` into the wrapper. if I modified it the way you described I'd need to destruct the config within the index and do some logic on it rather than just passing it around>